src/NHapi.Base/Model/AbstractGroup.cs

```diff
@@ -527,7 +522,7 @@ private IStructure TryToInstantiateStructure(Type c, string name)
                         var argClasses = new[] { typeof(IGroup), typeof(IModelClassFactory) };
                         var argObjects = new object[] { this, myFactory };
                         var con = c.GetConstructor(argClasses);
-                        o = con.Invoke(argObjects);
+                        o = con!.Invoke(argObjects);
```

Comment:
thinking about it... could you remove the `try...catch` and refactor to `o = c.GetConstructor(argClasses)?.Invoke(argObjects) ?? Activator.CreateInstance(c)` ?